rz-cmn-srp: Update uEnv.txt and platform_info.json for uboot integration

ThanhNguyenRVC opened the pull request:
Change includes:
- Remove hard-coded U-Boot env from uEnv.txt
- Update `platform_info.json` to include uboot information
- Support `rzv2h-evk-ver1` platform

ThanhNguyenRVC committed:
rz-cmn-srp: Update uEnv.txt and platform_info.json for uboot integration

Change includes:

- Remove hard code uboot env from uEnv.txt
- Update platform_info.json to include uboot information
- Support to build platform binary with revisions

Signed-off-by: Thanh Nguyen <thanh.nguyen.wx@bp.renesas.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

vudangRVC commented on recipes-tools/binmake/files/platform_info.json:
```diff
@@ -107,11 +107,40 @@
+        "kernel_dtb_desc": []
+    },
+
+    "rzv2h-evk-ver1": {
```
rzv2h-evk-ver1 is still a rzv2h-evk board but version 1. So, you should keep the name "rzv2h-evk" and change the revision instead.

For now, we have two boards. Let's change the revision based on the CHIP version. I'm not sure what the different names for the two boards are at the moment.

tiennguyenuh replied:
I'll use the same board name but with different revisions. As a result, the platform binary files will be named like:

rzv2h-evk-ver<revision_number>-platform-settings.bin

Build output
```
rzv2h-evk-ver1.20-platform-settings.bin
rzv2h-evk-ver1.20-platform-settings.srec
rzv2h-evk-ver1.21-platform-settings.bin
rzv2h-evk-ver1.21-platform-settings.srec
```

vudangRVC replied:
Let's use 1.00 and 2.00 for now.

tiennguyenuh replied:
I've updated this.

vudangRVC commented on recipes-tools/binmake/platform-gen.bb:
```diff
@@ -20,6 +20,12 @@ do_compile () {
         objcopy -I binary -O srec --adjust-vma=0x0000 --srec-forceS3 ${S}/${target}-platform-settings.bin ${S}/${target}-platform-settings.srec
     done
+
+    # Support special type of rzv2h
```
The binmake tool should include a revision option to differentiate between the two board versions

tiennguyenuh replied:
Add the --all-revisions argument to build binaries for all board revisions.

tiennguyenuh committed:
rz-cmn: update revision numbers and uEnv.txt file

Changes include:
	- Update revision numbers for all supported board.
	- Replace the long elif chain for DTB selection with a
          U-Boot command.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

vudangRVC commented on recipes-extend/uenv/files/uEnv.txt:
```diff
@@ -6,9 +6,29 @@
-        mmc_fdt=if test "${board_id}" = "rzg2l-sbc"; then setenv fdtfile rzg2l-sbc.dtb; elif test "${board_id}" = "rzg2l-evk"; then setenv fdtfile r9a07g044l2-smarc.dtb; elif test "${board_id}" = "rzv2l-evk"; then setenv fdtfile r9a07g054l2-smarc.dtb; elif test "${board_id}" = "rzv2h-evk"; then setenv fdtfile r9a09g057h4-evk-ver1.dtb; fi; fatload mmc 0:1 0x48000000 dtb/renesas/${fdtfile}
-        mmc_ovfdt=if run mmc_fdt; then fdt addr ${dtb_addr}; fdt resize 0x10000; if env exists enable_overlay_i2c &&    test ${enable_overlay_i2c} = 1 -o ${enable_overlay_i2c} = yes ; then echo Applying DT overlay: rzg2l-sbc-ext-i2c.dtbo ;fatload mmc 0:1 ${dtbo_addr} ${mmc_overlays}/rzg2l-sbc-ext-i2c.dtbo; fdt addr ${dtb_addr}; fdt apply ${dtbo_addr}; fi; if env exists enable_overlay_spi &&    te ...
+        mmc_image=fatload mmc ${mmcdev}:${mmcpart} ${image_addr} Image
+        mmc_ovfdt=if run mmc_fdt; then fdt addr ${dtb_addr}; fdt resize 0x10000; if env exists enable_overlay_i2c &&    test ${enable_overlay_i2c} = 1 -o ${enable_overlay_i2c} = yes ; then echo Applying DT overlay: rzg2l-sbc-ext-i2c.dtbo ;fatload mmc ${mmcdev}:${mmcpart} ${dtbo_addr} ${mmc_overlays}/rzg2l-sbc-ext-i2c.dtbo; fdt addr ${dtb_addr}; fdt apply ${dtbo_addr}; fi; if env exists enable_ove ...
```
Please use newline conjunctions to make maintenance easier.

tiennguyenuh replied:
I’ve updated this and tested all options

<img width="1500" height="655" alt="image" src="https://github.com/user-attachments/assets/70621584-c053-4cb4-9b63-0447bc90b8e9" />

vudangRVC replied:
I don't want users to see a lot of code in a configuration file. Can you move the code section to another file?

tiennguyenuh replied:
I've updated this. Relevant PR is in u-boot: https://github.com/Renesas-SST/u-boot/pull/13
